code/modules/antagonists/bloodsuckers/bloodsucker_mobs.dm

```diff
@@ -0,0 +1,178 @@
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab = 5)
+	obj_damage = 50
+	environment_smash = ENVIRONMENT_SMASH_WALLS
+	speak_emote = list("gnashes")
+	var/satiation = 0
+
+/mob/living/simple_animal/hostile/bloodsucker/giantbat
+	name = "giant bat"
+	desc = "That's a fat ass bat."
+	icon_state = "batform"
+	icon_living = "batform"
+	icon_dead = "bat_dead"
+	icon_gib = "bat_dead"
+	maxHealth = 500
+	health = 500
+	attacktext = "bites"
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab = 3)
+	attack_sound = 'sound/weapons/bite.ogg'
+	obj_damage = 35
+	pass_flags = PASSTABLE | PASSCOMPUTER
+	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
+	movement_type = FLYING
+	speak_emote = list("loudly squeaks")
+
+/mob/living/simple_animal/hostile/bloodsucker/possessedarmor
+	name = "possessed armor"
+	desc = "Whatever possessed this suit of armor to suddenly walk about and start killing everyone?"
+	icon_state = "posarmor"
+	icon_living = "posarmor"
+	response_help = "pokes"
+	response_disarm = "pushes"
+	response_harm = "punches"
+	maxHealth = 250
+	health = 250
+	attacktext = "rends"
+	obj_damage = 50
+	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
+	speak_emote = list("manifests")
+	var/upgraded = FALSE
+
+/mob/living/simple_animal/hostile/bloodsucker/possessedarmor/upgraded
+	name = "armed possessed armor"
+	icon_state = "posarmor_sword"
+	icon_living = "posarmor_sword"
+	upgraded = TRUE
+	obj_damage = 55
+	harm_intent_damage = 25
+	melee_damage_lower = 25
+	melee_damage_upper = 25
+
+///////////////////////////////
+///      Inheritances       ///
+///////////////////////////////
+
+/mob/living/simple_animal/hostile/bloodsucker/Destroy() //makes us alive again
+	if(bloodsucker && mind)
+		visible_message(span_warning("[src] rapidly transforms into a humanoid figure!"), span_warning("You forcefully return to your normal form."))
+		playsound(src, 'sound/weapons/slash.ogg', 50, 1)
+		if(mind)
+			mind.transfer_to(bloodsucker)
+		bloodsucker.forceMove(get_turf(src))
+		if(istype(src, /mob/living/simple_animal/hostile/bloodsucker/werewolf) && istype(src, /mob/living/simple_animal/hostile/bloodsucker/possessedarmor))
+			STOP_PROCESSING(SSprocessing, src)
+		return ..()
+
+/mob/living/simple_animal/hostile/bloodsucker/death()
+	if(bloodsucker && mind)
+		mind.transfer_to(bloodsucker)
+		bloodsucker.death()
+	if(src)
+		addtimer(CALLBACK(src, .proc/gib), 20 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/bloodsucker/proc/devour(mob/living/target)
+	if(istype(src, /mob/living/simple_animal/hostile/bloodsucker/werewolf))
+		var/mob/living/simple_animal/hostile/bloodsucker/werewolf/ww = src
+		ww.satiation++
+		src.visible_message(span_danger("[src] devours [target] whole!"), \
+		span_userdanger("You devour [target] and feel the beast inside you sedate itself a little, you'll need to feast [3 - ww.satiation] more times to become human again."))
+	health += target.maxHealth / 4
+	target.gib()
```

Comment:
might want to do the thing marrow weavers do where they eat all your organs instead since gibs are nasty